int32[22] hardness #numOfJoints
